hydroflow_lang/src/graph/ops/mod.rs

```diff
@@ -580,6 +580,35 @@ pub const OPERATORS: [OperatorConstraints; 24] = [
             }
         }),
     },
+    OperatorConstraints {
+        name: "unique",
+        hard_range_inn: RANGE_1,
+        soft_range_inn: RANGE_1,
+        hard_range_out: RANGE_1,
+        soft_range_out: RANGE_1,
+        ports_inn: None,
+        ports_out: None,
+        num_args: 0,
+        input_delaytype_fn: &|_| Some(DelayType::Stratum),
+        write_fn: &(|&WriteContextArgs { op_span, .. },
+                     &WriteIteratorArgs {
+                         ident,
+                         inputs,
+                         arguments,
+                         is_pull,
+                         ..
+                     }| {
+            assert!(is_pull);
+            let input = &inputs[0];
+            let write_iterator = quote_spanned! {op_span=>
+                let #ident = #input.fold(HashSet::new(#arguments), |mut prev, nxt| {prev.insert(nxt); prev}).into_iter();
```

Comment:
This will only deduplicate for a given execution, I'm guessing we would want it to be for an epoch or forever (across epochs). Would need to use the state API.